examples/samples_test.go

```diff
@@ -0,0 +1,41 @@
+package examples
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"testing"
+	"time"
+)
+
+func TestRunSamples(t *testing.T) {
+	items, _ := os.ReadDir(".")
+	for _, item := range items {
+		if item.IsDir() {
+			fmt.Printf("Running sample %v\n", item.Name())
```

Comment:
The samples (currently) aren't designed to run in parallel, as each of them start the emulator on the fixed port 9010. Running them in parallel could theoretically also cause resource issues, depending on the number of tests running in parallel, as each of them start the emulator, which requires a non-trivial amount of memory. So I'd prefer to keep these tests sequential to avoid any possible side-effects of running them in parallel.

(The `fmt.Printf(..)` was just for debugging, as running the samples did not work in all cases, but that's partly because GitHub Actions does not support running a Docker container in this fashion on any other platform than Ubuntu)

Reply:
Ah, in that case using `t.Run` without `t.Parallel()` is completely fine as well.

`t.Run` helps to separate the test output and if one of them fails, it'll still check whether anything else fails.